src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -27,19 +27,24 @@ def main():
     args = parser.parse_args(command_args)
     port = args.port
 
-    if port and is_port_in_use():
+    if port and is_port_in_use(port):
         raise UserErrorException(f"Service port {port} is used.")
     if not port:
-        # Read and write permission for user.
-        mode = stat.S_IRUSR | stat.S_IWUSR
-        (HOME_PROMPT_FLOW_DIR / SERVICE_CONFIG_FILE).touch(mode=mode, exist_ok=True)
+        (HOME_PROMPT_FLOW_DIR / SERVICE_CONFIG_FILE).touch(mode=read_write_by_user(), exist_ok=True)
         with open(HOME_PROMPT_FLOW_DIR / SERVICE_CONFIG_FILE, "r") as f:
-            service_config = yaml.safe_load(f)
+            service_config = yaml.safe_load(f) or {}
             port = service_config.get("service", {}).get("port", None)
+        if not port:
+            with open(HOME_PROMPT_FLOW_DIR / SERVICE_CONFIG_FILE, "w") as f:
```

Comment:
maybe we should have another file store the actual port. 
e.g. ~/.promptflow/pfs.port

Reply:
change to pfs.port